File: Packages/Yage/Sources/Yage/Capabilities/AnimationsProvider.swift
import Foundation

open class AnimationsProvider: Capability {
    open func randomAnimation() -> EntityAnimation? {
        subject?.species.behaviors
            .filter { $0.trigger == .random }
            .flatMap { $0.possibleAnimations }
            .randomElement()
    }
}

extension Entity {
    var animationsProvider: AnimationsProvider? {
        capability(for: AnimationsProvider.self)
    }
}

File: Sources/windows/yage/capabilities/animations_provider.py
from random import choice
from yage.models.animations import EntityAnimation
from yage.models.capability import Capability


class AnimationsProvider(Capability):
    def __init__(self, subject):
        super().__init__(subject)
        self.subject.animations_provider = self

    def random_animation(self) -> EntityAnimation:
        return choice(self.subject.species.animations)
